components/hab/src/command/plan/init.rs

```diff
@@ -107,19 +92,11 @@ pub fn start(ui: &mut UI,
     }
 
     // We want to render the configured variables.
-    if windows {
-        if with_all || scaffold.is_none() {
-            let rendered_plan = handlebars.template_render(FULL_PLAN_PS1_TEMPLATE, &data)?;
-            create_with_template(ui, &format!("{}/plan.ps1", root), &rendered_plan)?;
-        } else {
-            let rendered_plan = handlebars.template_render(DEFAULT_PLAN_PS1_TEMPLATE, &data)?;
-            create_with_template(ui, &format!("{}/plan.ps1", root), &rendered_plan)?;
-        }
-    } else if with_all || scaffold.is_none() {
-        let rendered_plan = handlebars.template_render(FULL_PLAN_TEMPLATE, &data)?;
-        create_with_template(ui, &format!("{}/plan.sh", root), &rendered_plan)?;
+    if cfg!(windows) {
```

Comment:
This means that we can only create Windows plans on Windows machines, and Linux plans on Linux machines, corret? That is probably a bit too strict. There may not be a lot of Linux users that want to make Windows plans, but our (enterprise) Windows users often also have Linux services they run.

Granted, those same Windows users could easily drop into WSL for most of their Linux needs. 

If it's not too difficult, it'd be nice to have a way for users to create whatever kind of plan they need to, regardless of whatever platform they happen to currently be on. It should probably default to the platform they're currently on, but still be modifiable.

That being said, I'm happy to leave that for a subsequent PR until we can really nail down the UX of this work.

Reply:
That is correct. There used to be a `--windows` flag that forced a powershell template, but I removed that. I made a separate issue (#6516) to discuss the UX. If we come to a decision on that I can go ahead and integrate it into this PR.

Reply:
I'm inclined to take a [YAGNI](https://en.wikipedia.org/wiki/You_aren%27t_gonna_need_it) approach on this one. It's always easy to add this later, but much harder to remove support once we've added it. I'm willing to bet the development of a plan for an arch other than the one currently running is more of an imagined use case than a reality.

Reply:
@davidMcneil Ah, cool, you've got a separate issue.

Let's keep the current PR as it is, and if that issue attracts any additional input, we can revisit later.